sdk-src/src/test/java/com/hpe/adm/nga/sdk/network/google/TestGoogleHttpClient.java

```diff
@@ -75,4 +93,95 @@ public void testRequestRetry() throws Exception {
                 times(GoogleHttpClient.getHttpRequestRetryCount() + 1))
                 .invoke("execute", any(), anyInt());
     }
+
+
+    @Test
+    public void testParallelRequestRetry() throws InterruptedException {
+        ConfigurationProperties.logLevel("WARN");
+
+        ClientAndServer clientAndServer = startClientAndServer();
+        Octane octane;
+        long totalExecutionTime = 500;
+        long cookieExpirationTime = 250;
+
+        setServerResponse(clientAndServer, cookieExpirationTime);
+        Authentication authentication = new SimpleUserAuthentication("", "");
+        String url = "http://localhost:" + clientAndServer.getLocalPort();
+        GoogleHttpClient spyGoogleHttpClient = spy(new GoogleHttpClient(url));
+        octane = new Octane.Builder(authentication, spyGoogleHttpClient).Server(url).workSpace(1002).sharedSpace(1001).build();
+
+        int nrCores = Runtime.getRuntime().availableProcessors();
```

Comment:
can be implemented nicer with the java streams api:
IntStream.rangeClosed(1, cores).parallel().forEach(LAMBDA)
No need to join it, no need to keep an array of threads